runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/work/ProcessWorkItemClient.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.runners.dataflow.worker.windmill.work;
+
+import com.google.auto.value.AutoValue;
+import org.apache.beam.runners.dataflow.worker.windmill.Windmill.WorkItem;
+import org.apache.beam.runners.dataflow.worker.windmill.WindmillStream.CommitWorkStream;
+import org.apache.beam.runners.dataflow.worker.windmill.WindmillStream.GetDataStream;
+
+/**
+ * A client context to process {@link WorkItem} and route all subsequent Windmill WorkItem API calls
+ * to the same backend worker. Wraps the {@link WorkItem}.
+ */
+@AutoValue
+public abstract class ProcessWorkItemClient {
+  public static ProcessWorkItemClient create(
+      WorkItem workItem, GetDataStream getDataStream, CommitWorkStream commitWorkStream) {
+    return new AutoValue_ProcessWorkItemClient(workItem, getDataStream, commitWorkStream);
+  }
+
+  /** {@link WorkItem} being processed. */
+  public abstract WorkItem workItem();
+
+  public abstract GetDataStream getDataStream();
```

Comment:
The GetDataStream that connects to windmill worker handling the work item.

Reply:
done